lib/src/realm_class.dart

```diff
@@ -562,6 +562,24 @@ class Realm implements Finalizable {
 
     realmCore.writeCopy(this, config);
   }
+
+  /// Update the `Realm` instance and outstanding objects to point to the most recent persisted version.
+  ///
+  /// If another process or thread has made changes to the realm file, this causes
+  /// those changes to become visible in this realm instance.
+  /// Typically you don't need to call this method since Realm has auto-refresh built-in.
+  /// Returns `true` only if there are changed data to refresh.
```

Comment:
This drastically differs from what .NET has for example. Here is the doc API .NET has currently
https://www.mongodb.com/docs/realm-sdks/dotnet/latest/reference/Realms.Realm.html#Realms_Realm_Refresh

> Note that this may return true even if no data has actually changed.

Also this is not what the native code is doing. Native code is returning `did_refresh` and it does not return `true` if any data actually changed